x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -512,6 +512,8 @@ public static boolean verifyHash(SecureString data, char[] hash) {
             PBEKeySpec keySpec = new PBEKeySpec(data.getChars(), salt, cost, PBKDF2_KEY_LENGTH);
             result.put(Base64.getEncoder().encodeToString(secretKeyFactory.generateSecret(keySpec).getEncoded()));
             return result.array();
+        } catch (AssertionError ae){
```

Comment:
My thinking is that we should scope this tightly first and expand if/when we get reports/CI failures for additional errors. The ones we know are thrown are all `AssertionError` and catching all `Error` felt too wide of a net. Not holding a strong opinion on this, happy to discuss

Reply:
Somehow missed this comment in the previous pass ... I see you already considered it. I suggest a wider net (even wider then `Error`) because 1) any future fix to this would require a release which is purely overhead; 2) I cannot think of a valid reason for ES to shutdown here due to any kinda of error. So wider net (at least `Error`) should provide mostly benefit.

Reply:
Thanks for the comment Yang. I see your points. I'll let Lyudmila weigh in too and we can reach a consensus

Reply:
I would leave an ability to throw and shutdown for when something unthinkable happens. Catching `Error` should be ok though